WebNetSample/Controllers/ProductsController.cs

```diff
@@ -18,4 +23,43 @@ public async Task<IActionResult> Index()
 
         return View(productsWithDetails);
     }
+
+    public async Task<IActionResult> AllProducts()
+    {
+        var productsWithDetails = await _productService.GetListAsync(_paginationParameters);
+
+        return View(productsWithDetails);
+    }
+
+    [HttpGet]
+    public IActionResult Add()
+    {
+        return View();
+    }
+
+    [HttpPost]
+    public async Task<ActionResult> Add(Product product)
+    {
+        await _productService.AddAsync(product);
+
+        return RedirectToAction("Index");
+    }
+
+    [HttpGet]
+    public IActionResult Edit(Guid id)
+    {
+        var productToEditTask = _productService.GetByIdAsync(id);
```

Comment:
Why don't you wrap your action with `Task<T>` class and put this operation into `await` step?

Reply:
Sorry, my bad. I'll change it as you mentioned